src/libYARP_math/include/yarp/math/FrameTransform.h

```diff
@@ -17,119 +17,124 @@
 
 namespace yarp
 {
-    namespace math
+namespace math
+{
+
+class YARP_math_API FrameTransform
+{
+public:
+
+    std::string parentFrame;
```

Comment:
The use of four data members is terribly confusing and its compatibility with ROS is suspicious
We are now confusing different concepts.
A frame transform defines a transformation from a source to a destination frame.
A frame can have a name and a parent. (NB: In ROS frames have child frames, not parent frames)
see: http://wiki.ros.org/tf/Overview/Data%20Types

Reply:
ok, i forgot to add some comments there. The two new member are supposed to replace the old ones which have to be deprecated (in the new implementation they are not used). and this avoid the confusion that created #1775.

regarding the parent/child i knew the difference with ROS and find very wrong to express the child frameID and not the parent. to explain the reason why just follow the following statement:

- the preferable structure for expressing position and orientation between entities, if we have to 
choose between a Tree and graph is a **Tree** in order to avoid conflict
- in a tree a child can have only one parent and a parent can have multiple child
- the **ID** of somethig is supposed to be unique and rappresent the key of the data. so the frameID should be unique in our structure
- with a frameID unique enforcement (via map structure) we can assure the presence of only one parent per frame thus enforcing the concistence of the tree.

conclusion: the tf in this PR are more suitable to be organized in a proper structure.

at the moment i find the ROS way to rappresent that structure wrong. **maybe (very probable) i am missing something important. so please enlighten me.**

after saying that, if we want to stay consistent with ROS despite it to be wrong(possibly) i have no problem. I only want to confront this design and understand why it is not better.

Reply:
on the other way if you want a consistent structure you have to enforce the unique-ness of the childFrame mamber which is counterintuitive

Reply:
Unfortunately, I cannot test ROS compatibility, because your branch is based on a broken YARP. Please rebase or integrate at least aee548794d6b ee013be596d4c63f.

Reply:
i've rebased it on devel on 18/03